### PR TITLE
Added: Paragraph Selector & Model Selector

### PR DESCRIPTION
**Pull request by gptjozef**
Added OpenAI placeholder to env example

### CODE REVIEW

**Comment by gptjozef** [comment]
Summary of Changes

This pull request introduces significant enhancements to both the frontend and backend, aimed at providing users with the ability to select the desired OpenAI model and specify the number of paragraphs for script generation. These changes are designed to give users more control over the content creation process and to address issues with excessive video length due to too many paragraphs being generated. Backend Changes (Backend/gpt.py)

Enhanced generate_script Functionality Modified the generate_script function to respect the number of paragraphs specified by the user, preventing the generation of overly lengthy scripts.

AI Model Selection Updated the generate_response and related functions to accommodate the selection of different AI models, allowing for flexibility in choosing between models like GPT-3.5 and GPT-4.

Frontend Changes

AI Model Selection UI Implemented a new feature in the frontend that allows users to select from available OpenAI models. This selection is then passed to the backend to determine which model is used for content generation.

Paragraph Selector Added a paragraph selector to the frontend, giving users the ability to define the number of paragraphs they want to generate for their video scripts.

Testing and Validation
Performed end-to-end testing to ensure that the frontend selections for AI models and paragraph numbers are correctly passed to and handled by the backend. Validated that the generated video scripts adhere to the user's specifications regarding length and content.

Additional Notes
Reviewers should verify that the new frontend controls are intuitive and that the backend correctly processes the inputs. It is important to ensure that the changes maintain compatibility with any existing features and that they can be extended to other language models in the future.

**Comment by gptjozef** [comment]
please reject my twitter callback on this pull lol i don't know how to not include that

**Comment by Arne1303** [comment]
Those files in `__pycache__` really shouldn't be there, can you delete the whole folder? Also, it looks like you changed the line endings in a lot of files this will make this hard to merge with other changes, maybe you can fix that two before this will be reviewed.


@FujiwaraChoki I will make a pr to make sure that the cache in sub folders are also excluded in the .gitignore file, they are easy to generated by simply starting python from the wrong folder.

**Comment by FujiwaraChoki** [comment]
Amazing PR, really. I will merge once the issues mentioned by Arne are solved, and I have tested everything. Thank you!

File: Backend/gpt.py [hunk 1]
```diff
@@ -1,172 +1,204 @@
...
+        str: The response from the AI model.
+    """
+    if ai_model == 'g4f':
+        response = g4f.ChatCompletion.create(
+            model=g4f.models.gpt_35_turbo_16k_0613,
+            messages=[{"role": "user", "content": prompt}],
+        )
+    elif ai_model in ['gpt3.5-turbo', 'gpt4']:
+        model_name = "gpt-3.5-turbo" if ai_model == 'gpt3.5-turbo' else "gpt-4-1106-preview"
+        response = openai.ChatCompletion.create(
```
Comment (zanmat0) [inline Backend/gpt.py:35]: ChatCompletion is deprecated in newer versions of _openai_. 
Change line to 
`response = openai.chat.completions.create(`
as described in the official docs: https://platform.openai.com/docs/quickstart?context=python

**Comment by RicardoMonteiroSimoes** [comment]
probably best to add `__pycache__` to `.gitignore`

File: Backend/gpt.py [hunk 2]
```diff
@@ -1,172 +1,204 @@
...
+from dotenv import load_dotenv
+import os
+
+# Load environment variables
+load_dotenv("../.env")
+
+# Set environment variables
+OPENAI_API_KEY = os.getenv('OPENAI_API_KEY')
+openai.api_key = OPENAI_API_KEY
+
```
Comment (RicardoMonteiroSimoes) [inline Backend/gpt.py:16]: Should probably check if it's there or not, and set some variable to true/false so we can safely abort anything that tries to run it

**Comment by evolite** [comment]
The amount of paragraphs could be changed to amount of sentences. As i find that a pragraph for a short is allready too much.

**Comment by RicardoMonteiroSimoes** [comment]
> The amount of paragraphs could be changed to amount of sentences. As i find that a pragraph for a short is allready too much.

Agree here, I've had great success with the following:

```
The script is to be returned as a string. Keep it concise, so a maximum of 5 sentences.
```

(See PR https://github.com/FujiwaraChoki/MoneyPrinter/pull/120/files, but could be added here)

**Comment by gptjozef** [comment]
> Amazing PR, really. I will merge once the issues mentioned by Arne are solved, and I have tested everything. Thank you!

> Love It! But those files in `__pycache__` really shouldn't be there, can you delete the whole folder? Honestly not sure how you added the files, all `__pycache__` folders should be ignored by git. Also, it looks like you changed the line endings in a lot of files this will make this hard to merge with other changes, maybe you can fix that two before this will be reviewed.

Hey I'm so happy you all liked my changes. I am new to OSS contributions, this is my first. I'm kinda *redacted* when it comes to this hence my patchwork changes I made to comply with these requests. I don't know how I messed up the formatting so I just copied your code over again, and manually made my edits. I did compile and test on my end and everything does appear to still work. Got a video done with the correct AI model and paragraphs selected. Enjoy, so happy I could contribute in less than 24 hours since using this project.

I have a couple more ideas I'll hopefully get to soon ;) 

Cheers, 
Joe

**Pull request merged**